kernel/src/hil/aes.rs

```diff
@@ -0,0 +1,14 @@
+use returncode::ReturnCode;
+
+pub trait AESDriver {
+    fn init(&self);
+    fn set_key(&self, key: &'static mut [u8], len: u8);
```

Comment:
The key is a known length for any given implementation. Is there a way we can encode this (like the time HIL does for frequency) so the user cannot pass in an arbitrary length key?